src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === undefined && i === this.length - 1)
+      || (this[i] === null && i === this.length - 1)) {
```

Comment:
Why do you need these checks `i === this.length - 1`? 
I think you don't need whole this block.

Reply:
(i === this.length - 1) I check if it element is the last. If it is true I don't add separator